engine/apps/alerts/models/alert_group.py

```diff
@@ -1042,31 +1032,24 @@ def bulk_acknowledge(user: User, alert_groups: "QuerySet[AlertGroup]") -> None:
             send_alert_group_signal.apply_async((log_record.pk,))
 
     @staticmethod
-    def bulk_resolve(user: User, alert_groups: "QuerySet[AlertGroup]") -> None:
-        AlertGroupLogRecord = apps.get_model("alerts", "AlertGroupLogRecord")
-
-        # stop maintenance for maintenance incidents
-        alert_groups_to_stop_maintenance = alert_groups.filter(resolved=False, maintenance_uuid__isnull=False)
-        for alert_group in alert_groups_to_stop_maintenance:
-            alert_group.stop_maintenance(user)
-
-        root_alert_groups_to_resolve = alert_groups.filter(
-            resolved=False,
+    def bulk_acknowledge(user: User, alert_groups: "QuerySet[AlertGroup]") -> None:
+        root_alert_groups_to_acknowledge = alert_groups.filter(
+            ~Q(acknowledged=True, resolved=False),  # don't need to ack acknowledged incidents once again
             root_alert_group__isnull=True,
-            maintenance_uuid__isnull=True,
+            maintenance_uuid__isnull=True,  # don't ack maintenance incident
         )
-        if root_alert_groups_to_resolve.count() == 0:
-            return
-
-        organization = root_alert_groups_to_resolve.first().channel.organization
-        if organization.is_resolution_note_required:
-            root_alert_groups_to_resolve = root_alert_groups_to_resolve.filter(
-                Q(resolution_notes__isnull=False, resolution_notes__deleted_at=None)
-            )
-        dependent_alert_groups_to_resolve = AlertGroup.all_objects.filter(
-            root_alert_group__in=root_alert_groups_to_resolve
+        # Find all dependent alert_groups to update them in one query
+        # convert qs to list to prevent changes by update
+        root_alert_group_pks = list(root_alert_groups_to_acknowledge.values_list("pk", flat=True))
+        dependent_alert_groups_to_acknowledge = AlertGroup.unarchived_objects.filter(
+            root_alert_group__pk__in=root_alert_group_pks
         )
-        alert_groups_to_resolve = root_alert_groups_to_resolve | dependent_alert_groups_to_resolve
+        AlertGroup._bulk_acknowledge(user, root_alert_groups_to_acknowledge)
+        AlertGroup._bulk_acknowledge(user, dependent_alert_groups_to_acknowledge)
```

Comment:
we can probably run these two queries in a transaction so we don't have any inconsistencies between state of root and dependent alert groups